resources/kiali/templates/virtualservice.yaml

```diff
@@ -1,14 +1,16 @@
 apiVersion: networking.istio.io/v1alpha3
 kind: VirtualService
 metadata:
-  name: kiali-virtualservice
+  name: {{ template "kiali.name" . }}
   namespace: {{ .Release.Namespace }}
   labels:
     app: {{ template "kiali.name" . }}
-    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    chart: {{ template "kiali.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
 spec:
   gateways:
-  - {{ .Values.global.istio.gateway.name }}
+  - kyma-gateway
```

Comment:
Better to use a variable here as we have the name of the gateway set as variable also

Reply:
done